src/transformers/models/seamless_m4t/modeling_seamless_m4t.py

```diff
@@ -0,0 +1,4058 @@
+            decoder_head_mask=decoder_head_mask,
+            cross_attn_head_mask=cross_attn_head_mask,
+            encoder_outputs=encoder_outputs,
+            past_key_values=past_key_values,
+            inputs_embeds=inputs_embeds,
+            decoder_inputs_embeds=decoder_inputs_embeds,
+            labels=labels,
+            use_cache=use_cache,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+            **kwargs,
+        )
+
+    @torch.no_grad()
+    def generate(
+        self,
+        input_ids: Optional[torch.Tensor] = None,
+        **kwargs,
+    ) -> Union[str, torch.LongTensor]:  # TODO: output
+        kwargs_text_generation = {}
+        kwargs_speech_generation = {}
+        for key, value in kwargs.items():
+            if key.startswith("text_generation_"):
+                key = key[len("text_generation_") :]
+                kwargs_text_generation[key] = value
+            elif key.startswith("speech_generation_"):
+                key = key[len("speech_generation_") :]
+                kwargs_speech_generation[key] = value
+            else:
+                # If the key is already in a specific config, then it's been set with a
+                # submodules specific value and we don't override
+                if key not in kwargs_text_generation:
+                    kwargs_text_generation[key] = value
+                if key not in kwargs_speech_generation:
+                    kwargs_speech_generation[key] = value
+
+        kwargs_text_generation["output_hidden_states"] = True
+        kwargs_text_generation["return_dict_in_generate"] = True
+        kwargs_text_generation["output_scores"] = True
+
+        output_text = self.input_model.generate(input_ids, **kwargs_text_generation)
+        
+        batch_size = len(input_ids)
+        num_return_sequences = len(output_text.sequences) // batch_size
+        sequences = output_text.sequences
+
+        # compute last hidden state 
+        t2u_input_embeds = self.compute_last_hidden_states_per_sample(output_text.decoder_hidden_states, output_text.get("beam_indices", None))
+        
+        # take care of num_return_sequences
+        # take most probable hidden states per batch of return_sequences
+        # (batch_size*num_return_sequences, ...) -> (batch_size,...)
+        if num_return_sequences > 1:
+            idx_most_probable_sequences_per_batch = output_text.sequences_scores.view(batch_size, -1).argmax(-1)
+            idx_most_probable_sequences_per_batch = idx_most_probable_sequences_per_batch + torch.arange(batch_size)*num_return_sequences
+            t2u_input_embeds = t2u_input_embeds[idx_most_probable_sequences_per_batch]
+            sequences = sequences[idx_most_probable_sequences_per_batch]
+
+        # TODO: is it the proper way, what's the priority with generation config and so on?
+        pad_token_id = self.config.pad_token_id
```

Comment:
I guess here we should look in `self.input_model.generation_config.pad_token_id`, since the `.generate` method will pad with these tokens